### ISSUE TITLE
info: Change path of core_info.cache

### ISSUE THREAD

**Issue by RobLoach** [libretro/RetroArch#12669]
## Description

In some distributions of RetroArch, the assets and info file directories is unwrittable. The Ubuntu PPAs, and Flatpak, for instance, distribute the files so that they are not inadvertently changed. Flatpak distributes `/app/share/libretro/info/`, and makes it unwritable. 

When running RetroArch, the .info cache file attempts to get saved to `libretro/info/core_info.cache`, and fails, because the directory was distributed, and meant to be unwritable.

```
[ERROR] [Core Info] Failed to write to core info cache file: /app/share/libretro/info/core_info.cache
```

Would be great to have the cache info file written to a user's cache directory, like `~/.config/retroarch/cache` or something.

### Expected behavior

Have the `core_info.cache` file written to a user-space directory, like `~/.config/retroarch/cache/core_info.cache` instead.

### Actual behavior

It's currently writing to `/app/share/libretro/info/core_info.cache`, which ends up failing to write.

### Steps to reproduce the bug

1. Make your `info` directory unwritable
2. Run RetroArch
3. See the note in the logs for `[ERROR] [Core info] Failed to write core_info.cache`

### Bisect Results

https://github.com/libretro/RetroArch/pull/12397

### Version/Commit
You can find this information under Information/System Information

- RetroArch: https://github.com/libretro/RetroArch/commit/e5bbd9488f80694afdf40c19bedf492569f8d4f3

### Environment information

- OS: Ubuntu 20.04
- Compiler: gcc/Flatpak

**Comment by jdgleaver** [comment]
Yes, we'll need to sort this out - apparently it's also an issue on Vita, where the core info directory has to be read only...

Unfortunately it is *very* non-trivial (since it involves updating the frontend drivers for every platform, and we also have to figure out how to handle the core_info.refresh file...). I think as a short-term fix, we'll need to have RA automatically disable the core info cache the first time that a write fails - that will at least remove the performance overheads + error log.

**Comment by jdgleaver** [comment]
Okay, here's a band-aid that should tide us over until a new persistent cache directory can be set up: #12680 :)
